plugins/ltac/extratactics.ml4

```diff
@@ -975,6 +975,7 @@ TACTIC EXTEND unshelve
 | [ "unshelve" tactic1(t) ] ->
     [
       Proofview.with_shelf (Tacinterp.tactic_of_value ist t) >>= fun (gls, ()) ->
+      let gls = List.map Proofview_monad.with_empty_state gls in
```

Comment:
I don't think we should have to access `Proofview_monad` here, so `with_empty_state` should be reexported.